Core/Src/main.c

```diff
@@ -205,101 +209,103 @@ void clearReceivedData(ReceivedData* receivedData) {
 // Handles the interrupts of the different timers.
 void HAL_TIM_PeriodElapsedCallback(TIM_HandleTypeDef *htim)
 {
-	if(htim->Instance == htim6.Instance){
-		geneva_Update();
-	}
-	else if(htim->Instance == htim7.Instance) {
-		/* SQUARE WITH 90 DEGREES TURNS AT SIDES
-	  	float velocityRef[3];
-		velocityRef[0] = 0.0;
-		velocityRef[1] = 0.0;
-		velocityRef[2] = 0.0*M_PI;
-		halt = false;
-		static uint velTimer;
-		int reps = 1;
-		static int count = 0;
-		float v = 0.5;
-		int t = 1500;
-		if (HAL_GetTick() < 7000) {
-			velTimer = HAL_GetTick();
-		} else if (HAL_GetTick() - velTimer < t) {
-			velocityRef[body_x] = v;
-			velocityRef[body_y] = 0.0;
-			velocityRef[body_w] = 0.0;
-		} else if (HAL_GetTick() - velTimer < 2*t) {
-			velocityRef[body_x] = v;
-			velocityRef[body_y] = 0.0;
-			velocityRef[body_w] = 0.5*M_PI;
-		} else if (HAL_GetTick() - velTimer < 3*t) {
-			velocityRef[body_x] = 0.0;
-			velocityRef[body_y] = v;
-			velocityRef[body_w] = 0.0;
-		} else if (HAL_GetTick() - velTimer < 4*t) {
-			velocityRef[body_x] = 0.0;
-			velocityRef[body_y] = v;
-			velocityRef[body_w] = 0.5*M_PI;
-		} else if (HAL_GetTick() - velTimer < 5*t) {
-			velocityRef[body_x] = -v;
-			velocityRef[body_y] = 0.0;
-			velocityRef[body_w] = 0.0;
-		} else if (HAL_GetTick() - velTimer < 6*t) {
-			velocityRef[body_x] = -v;
-			velocityRef[body_y] = 0.0;
-			velocityRef[body_w] = 0.5*M_PI;
-		} else if (HAL_GetTick() - velTimer < 7*t) {
-			velocityRef[body_x] = 0.0;
-			velocityRef[body_y] = -v;
-			velocityRef[body_w] = 0.0;
-		} else if (HAL_GetTick() - velTimer < 8*t) {
-			velocityRef[body_x] = 0.0;
-			velocityRef[body_y] = -v;
-			velocityRef[body_w] = 0.5*M_PI;
-		} else if (count < reps-1) {
-			velTimer = HAL_GetTick();
-			count++;
-		} else {
-			velocityRef[body_x] = 0.0;
-			velocityRef[body_y] = 0.0;
-			velocityRef[body_w] = 0.0;
-		}
-
-
-		receivedData.stateRef[body_x] = velocityRef[body_x];
-		receivedData.stateRef[body_y] = velocityRef[body_y];
-		receivedData.stateRef[body_w] = velocityRef[body_w];
-		stateControl_SetRef(velocityRef);
-		*/
-
-
-		// State estimation
-		stateInfo.visionAvailable = receivedData.visionAvailable;
-		stateInfo.visionYaw = receivedData.visionYaw;
-		for (wheel_names wheel = wheels_RF; wheel <= wheels_LF; wheel++) {
-			stateInfo.wheelSpeeds[wheel] = wheels_GetState()[wheel];
+	if (xsens_CalibrationDone) {	// don't do anything until xsens calibration is done

```

Comment:
Can this be in the timer callback? Otherwise it can block stuff that should already be running. i.e. Shoot_Callback().